test-common/eos-test-convenience.c

```diff
@@ -46,6 +46,9 @@ etc_data_init (EtcData *data,
   data->server = NULL;
   data->subserver = NULL;
   data->client = NULL;
+  data->additional_directories_for_commit = NULL;
```

Comment:
Please split *additions* to the `test-common` code out from the commit which moves the existing code to `test-common` in the first place. It’s a lot easier to review huge diffs of code being moved from one place to another when you can be confident that none of the code is changing at the same time. (Well, the `Makefile.am`s will change, and some `#include` lines will change, but I wouldn’t expect changes to the actual test code like this.)

Reply:
Yeah, could be tricky. I'll spend some time doing that shortly.

Reply:
Okay, I split out quite a few changes based on the different things they added. Let me know if you'd like any to be squashed together.